prometheus-to-sd/translator/translator.go

```diff
@@ -42,28 +43,59 @@ var supportedMetricTypes = map[dto.MetricType]bool{
 
 const falseValueEpsilon = 0.001
 
-// TranslatePrometheusToStackdriver translates metrics in Prometheus format to Stackdriver format.
-func TranslatePrometheusToStackdriver(config *config.CommonConfig,
-	whitelisted []string,
-	metrics map[string]*dto.MetricFamily,
-	cache *MetricDescriptorCache) []*v3.TimeSeries {
+type timeSeriesBuilder struct {
+	config      *config.CommonConfig
+	whitelisted []string
+	cache       *MetricDescriptorCache
+	timeSeries  map[string]*v3.TimeSeries
+}
+
+func NewTimeSeriesBuilder(config *config.CommonConfig, whitelisted []string, cache *MetricDescriptorCache) *timeSeriesBuilder {
+	return &timeSeriesBuilder{
+		config:      config,
+		whitelisted: whitelisted,
+		cache:       cache,
+		timeSeries:  make(map[string]*v3.TimeSeries),
+	}
+}
+
+// Builds a new TimeSeries array and restarts the internal state.
+func (t *timeSeriesBuilder) Build() []*v3.TimeSeries {
+	s := make([]*v3.TimeSeries, 0, len(t.timeSeries))
+	for _, v := range t.timeSeries {
+		s = append(s, v)
+	}
+	t.timeSeries = make(map[string]*v3.TimeSeries)
+	return s
+}
 
+// Update translates metrics in Prometheus format to Stackdriver format and updates the current batch.
```

Comment:
Is there sense to translate metrics during every scrape? If we are going to send only 1 out 30 or 60 scrapes, then it's just a waste of CPU time. Propose to move this part of the logic to Build method and keep raw scraped metrics in timeSeriesBuilder.

Reply:
Good point. I'll change it.